app/routes/analyzer.tsx

```diff
@@ -65,6 +66,11 @@ export const handle: SendouRouteHandle = {
   navItemName: "analyzer",
 };
 
+// Resolves this Github issue: https://github.com/Sendouc/sendou.ink/issues/1053
+export const unstable_shouldReload: ShouldReloadFunction = ({ url }) => {
+  return Boolean(url) && !url.searchParams.get("weapon");
+};
```

Comment:
Good findings!! Should we just do:

```suggestion
export const unstable_shouldReload: ShouldReloadFunction = () => false;
```

since analyzer doesn't have a `loader` function set I believe this would be correct :) Although idk why Remix doesn't do this for us.

Reply:
Sure thing, let's do that instead then - I wasn't sure if we wanted to hard set it to `false` initially because I wasn't 100% on whether this function affects anything other than a loader, so thank you for checking.

> Although idk why Remix doesn't do this for us.

In the docs, they mention that they default the behaviour to reload everything to be safe: https://remix.run/docs/en/v1/api/conventions#unstable_shouldreload

> During client-side transitions, Remix will optimize reloading of routes that are already rendering, like not reloading layout routes that aren't changing. In other cases, like form submissions or search param changes, Remix doesn't know which routes need to be reloaded so it reloads them all to be safe. This ensures data mutations from the submission or changes in the search params are reflected across the entire page.